eng/pipelines/templates/steps/generate-nightly-branch-name.yml

```
@@ -0,0 +1,10 @@
+steps:
+  - pwsh: |
+      $branchName = if ($env:NIGHTLYBRANCHOVERRIDE) { 
+        $env:NIGHTLYBRANCHOVERRIDE
+      } else { 
+        $formattedDate = Get-Date -Format 'yyyy-MM-dd'
+        "nightly/$formattedDate"
+      }
```

While what you have works I find the below a little easier to understand and follow. I'm not generally a fan of assigning and if statement to a variable. 

```suggestion
      $branchName = $env:NIGHTLYBRANCHOVERRIDE
      if (!$branchName) { 
        $formattedDate = Get-Date -Format 'yyyy-MM-dd'
        $branchName"nightly/$formattedDate"
      }
```